File: Font-end/Admin Panel/admin-panel/src/app/data/models/product.ts
import { ProductDetail } from './product-detail';

export interface Product {
  id?: number;
  name: string;
  description: string;
  categoryId: number;
  categoryName?: string;
  isActive: number;
  rate?: number;
  rateSum?: number;
  details: ProductDetail[];
  images: string[];
}

File: Font-end/Mobile/AndroidNative/app/src/main/java/com/ltdt/coffeeshop_android_native/data/domains/Product.kt
package com.ltdt.coffeeshop_android_native.data.domains

import com.ltdt.coffeeshop_android_native.data.remote.ProductDTO

data class Product(
    val id: Int,
    val name: String,
    val categoryId: Int,
    val description: String,
    val categoryName: String,
    val isActive: Boolean,
    val rateSum: Long,
    val rate: Double,
    val details: List<ProductDetail>,
    val images: List<String>
)

fun Product.toProductDTO(): ProductDTO = ProductDTO(
    id = id,
    name = name,
    categoryId = categoryId,
    description = description,
    isActive = isActive,
    rate = rate,
    details = details.map { it.toProductDetailDTO() },
    images = images,
    categoryName = categoryName,
    rateSum = rateSum
)